commit f73d26228e00c7a768755d72f6a01e2fa80c5252
Author: Michael Dusan <michael.dusan@gmail.com>
Date:   Wed Jun 5 11:08:51 2024 -0400

    signal.c

diff --git a/linux-user/signal.c b/linux-user/signal.c
index 05dc4afb52..c07727dff0 100644
--- a/linux-user/signal.c
+++ b/linux-user/signal.c
@@ -37,6 +37,13 @@
 /* target_siginfo_t must fit in gdbstub's siginfo save area. */
 QEMU_BUILD_BUG_ON(sizeof(target_siginfo_t) > MAX_SIGINFO_LENGTH);
 
+#ifndef __SIGRTMIN
+#define __SIGRTMIN 32
+#endif
+#ifndef __SIGRTMAX
+#define __SIGRTMAX (NSIG-1)
+#endif
+
 static struct target_sigaction sigact_table[TARGET_NSIG];
 
 static void host_signal_handler(int host_signum, siginfo_t *info,
